Задание для кандидатов. Выполнить ревью кода.sql

```diff
@@ -1 +1,118 @@
+	-- Добавляем причину, по которой запись считается некорректной
+	select
+		cs.*
+		,case
+			when c.ID is null then 'UID клиента отсутствует в справочнике "Клиент"'
+			when c_dist.ID is null then 'UID дистрибьютора отсутствует в справочнике "Клиент"'
+			when s.ID is null then 'Сезон отсутствует в справочнике "Сезон"'
+			when cst.ID is null then 'Тип клиента отсутствует в справочнике "Тип клиента"'
+			when try_cast(cs.DateBegin as date) is null then 'Невозможно определить Дату начала'
+			when try_cast(cs.DateEnd as date) is null then 'Невозможно определить Дату окончания'
+			when try_cast(isnull(cs.FlagActive, 0) as bit) is null then 'Невозможно определить Активность'
+		end as Reason
+	into #BadInsertedRows
+	from syn.SA_CustomerSeasonal as cs
+	left join dbo.Customer as c on c.UID_DS = cs.UID_DS_Customer
+		and c.ID_mapping_DataSource = 1
+	left join dbo.Customer as c_dist on c_dist.UID_DS = cs.UID_DS_CustomerDistributor and c_dist.ID_mapping_DataSource = 1
+	left join dbo.Season as s on s.Name = cs.Season
+	left join syn.CustomerSystemType as cst on cst.Name = cs.CustomerSystemType
+	where c.ID is null
+		or c_dist.ID is null
+		or s.ID is null
+		or cst.ID is null
+		or try_cast(cs.DateBegin as date) is null
+		or try_cast(cs.DateEnd as date) is null
+		or try_cast(isnull(cs.FlagActive, 0) as bit) is null
+
+	-- Обработка данных из файла
+	merge into syn.CustomerSeasonal as cs
+	using (
+		select
+			cs_temp.ID_dbo_Customer
+			,cs_temp.ID_CustomerSystemType
+			,cs_temp.ID_Season
+			,cs_temp.DateBegin
+			,cs_temp.DateEnd
+			,cs_temp.ID_dbo_CustomerDistributor
+			,cs_temp.FlagActive
+		from #CustomerSeasonal as cs_temp
+	) as s on s.ID_dbo_Customer = cs.ID_dbo_Customer
+		and s.ID_Season = cs.ID_Season
+		and s.DateBegin = cs.DateBegin
+	when matched
+		and t.ID_CustomerSystemType <> s.ID_CustomerSystemType then
+		update
+		set ID_CustomerSystemType = s.ID_CustomerSystemType
+			,DateEnd = s.DateEnd
+			,ID_dbo_CustomerDistributor = s.ID_dbo_CustomerDistributor
+			,FlagActive = s.FlagActive
+	when not matched then
+		insert (ID_dbo_Customer, ID_CustomerSystemType, ID_Season, DateBegin, DateEnd, ID_dbo_CustomerDistributor, FlagActive)
+		values (s.ID_dbo_Customer, s.ID_CustomerSystemType, s.ID_Season, s.DateBegin, s.DateEnd, s.ID_dbo_CustomerDistributor, s.FlagActive);
+
+	-- Информационное сообщение
+	begin
+		select @ErrorMessage = concat('Обработано строк: ', @RowCount)
+		raiserror(@ErrorMessage, 1, 1)
+
+		--Формирование таблицы для отчетности
+		select top 100
+			bir.Season as 'Сезон'
+			,bir.UID_DS_Customer as 'UID Клиента'
+			,bir.Customer as 'Клиент'
+			,bir.CustomerSystemType as 'Тип клиента'
+			,bir.UID_DS_CustomerDistributor as 'UID Дистрибьютора'
+			,bir.CustomerDistributor as 'Дистрибьютор'
+			,isnull(format(try_cast(bir.DateBegin as date), 'dd.MM.yyyy', 'ru-RU'), bir.DateBegin) as 'Дата начала'
+			,isnull(format(try_cast(birDateEnd as date), 'dd.MM.yyyy', 'ru-RU'), bir.DateEnd) as 'Дата окончания'

```

Comment:
```suggestion
			,isnull(format(try_cast(bir.DateEnd as date), 'dd.MM.yyyy', 'ru-RU'), bir.DateEnd) as 'Дата окончания'

```
19. Отсутствует точка.